src/composables/queries/usePoolQuery.ts

```diff
@@ -79,6 +79,8 @@ export default function usePoolQuery(
       pool = await poolRepository.fetch(getQueryArgs());
     }
 
+    if (!pool) throw new Error('Pool does not exist');
```

Comment:
`await poolRepository.fetch(getQueryArgs());`  returns empty array if there is no pool or undefined?
shouldn't condition be like: `if (!pool?.length)`?

Reply:
The pool repository only returns a single pool, the individual fetchers return an array of pools.

Reply:
should we change return type of fetch function in poolRepository to
`public async fetch(queryArgs: GraphQLArgs): Promise<Pool | void> {` ?